### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by steveklabnik**
Building on top of my other branch, this also fixes all of your current warnings. Each part done in a separate commit.

### CODE REVIEW

**Comment by seth** [comment]
@steveklabnik Thank you for the patches!

We'll work on getting these integrated. This might take a minute as the team is attending ChefConf and we are doing internal development using Chef Delivery (and have yet to build the github integration :-P so we aren't in a spot where we can use the merge button just yet).

File: src/delivery/http/mod.rs [hunk 1]
```diff
@@ -189,7 +189,7 @@ impl APIClient {
     pub fn get(&self, path: &str) -> Result<HyperResponse, HttpError> {
         let url = self.api_url(path);
         let mut client = hyper::Client::new();
-        let req = client.get(url.as_slice());
+        let req = client.get(url.as_ref());
```
Comment (seth) [inline src/delivery/http/mod.rs]: Curious: why `.as_ref()` here instead of `&`?

Comment (steveklabnik) [inline src/delivery/http/mod.rs]: Due to generics, the Deref coercions don't appropriately kick in. At least, I'm pretty sure that's it :)

**Comment by steveklabnik** [comment]
@seth no worries! Also, I think the changes in the last day or two may require more, soooooo


Can't wait for beta. 


No worries if you can't take these patches for some reason :)

**Comment by tduffield** [comment]
This was addressed in afdec7b00ca93d933c5c251910397d251e18f12f

**Comment by steveklabnik** [comment]
:+1:

**Pull request closed without merging**